components/scream/scripts/gen_boiler.py

```diff
@@ -0,0 +1,1878 @@
+        lambda phys, sub, gb: re.compile(r".*;\s*$"), # end of struct decl
+        lambda *x           : "The cxx unit test struct declaration"
+    )),
+
+    ("cxx_bfb_unit_impl", (
+        lambda phys, sub, gb: "tests/{}_{}_tests.cpp".format(phys, sub),
+        lambda phys, sub, gb: create_template(phys, sub, gb, "cxx_bfb_unit_impl"),
+        lambda phys, sub, gb: get_cxx_close_block_regex(semicolon=True, at_line_start=True), # insert of end of struct
+        lambda phys, sub, gb: get_cxx_function_begin_regex("run_bfb", static=True),  # run_bfb
+        lambda phys, sub, gb: get_cxx_close_block_regex(comment="run_bfb"), # } // run_bfb # reqs special comment
+        lambda *x           : "The cxx bfb unit test implementation"
+    )),
+
+    ("cxx_eti", (
+        lambda phys, sub, gb: "{}_{}.cpp".format(phys, sub),
+        lambda phys, sub, gb: create_template(phys, sub, gb, "cxx_eti"),
+        lambda phys, sub, gb: re.compile(".*"), # insert at top of file
+        lambda phys, sub, gb: re.compile(".*"), # start at top of file
+        lambda phys, sub, gb: get_namespace_close_regex("scream"), #end of file
+        lambda *x           : "The cxx explicit template instatiation file"
+    )),
+
+    ("cmake_impl_eti", (
+        lambda phys, sub, gb: "CMakeLists.txt",
+        lambda phys, sub, gb: expect_exists(phys, sub, gb, "cmake_impl_eti"),
+        lambda phys, sub, gb: re.compile(r".*[)]\s*#\s*{} ETI SRCS".format(phys.upper())), # insert at end of ETI src list, reqs special comment
+        lambda phys, sub, gb: re.compile(r".*{}".format(get_piece_data(phys, sub, "cxx_eti", FILEPATH, gb))),
+        lambda phys, sub, gb: re.compile(".*"),
+        lambda *x           : "Make cmake aware of the ETI file if not cuda build"
+    )),
+
+    ("cmake_unit_test", (
+        lambda phys, sub, gb: "tests/CMakeLists.txt",
+        lambda phys, sub, gb: expect_exists(phys, sub, gb, "cmake_unit_test"),
+        lambda phys, sub, gb: re.compile(r".*[)]\s*#\s*{}_TESTS_SRCS".format(phys.upper())), # insert at end of test src list, reqs special comment
+        lambda phys, sub, gb: re.compile(r".*{}".format(os.path.basename(get_piece_data(phys, sub, "cxx_bfb_unit_impl", FILEPATH, gb)))),
+        lambda phys, sub, gb: re.compile(".*"),
+        lambda *x           : "Make cmake aware of the unit test"
+    )),
+
+])
+
+# physics map. maps the name of a physics packages containing the original fortran subroutines to:
+#   (path-to-origin, path-to-cxx-src)
+ORIGIN_FILE, CXX_ROOT, INIT_CODE = range(3)
+PHYSICS = {
+    "p3"   : (
+        "components/cam/src/physics/cam/micro_p3.F90",
+        "components/scream/src/physics/p3",
+        "p3_init();"
+    ),
+    "shoc" : (
+        "components/cam/src/physics/cam/shoc.F90",
+        "components/scream/src/physics/shoc",
+        "shoc_init(d.nlev, true);"
+    ),
+}
+
+# A good set of arg data for unit testing
```

Comment:
I'm not sure I understand what these arrays are for. I don't see them used in the file either, only in the function help. Unless I'm missing some subtle mechanism?

Reply:
After further reading the file, I _think_ these are only used in documentation examples. Might be worth saying it in the comment. Like "An example set of arg data to be used in the documentation of the methods below". So the user doesn't get confused about why `foo1` could ever be used in a real case (it got me thinking for a while).

Reply:
I'm a bit confused, did you see the comment above?
```
# A good set of arg data for unit testing
```
Are you just looking for a better comment?

Reply:
I saw it. But at first I thought those were actual data to be used in actual unit testing. But reading the rest of the file makes me think they are just a dummy set of arg data to be used in the comments section of each (python) function in the file, to provide example of the output produced by such functions. Is that right? If so, then yes, I'm looking for a better comment I guess.